ui/hooks/useNftCollectionsMetadata.ts

```diff
@@ -0,0 +1,64 @@
+import { useEffect, useMemo, useState } from 'react';
+import { NameType } from '@metamask/name-controller';
+import { Collection } from '@metamask/assets-controllers';
+import { useSelector } from 'react-redux';
+import { getCurrentChainId } from '../selectors';
+import { fetchNftCollectionsMetadata } from '../store/actions';
+
+export type UseNftCollectionsMetadataRequest = {
+  value: string;
```

Comment:
Should this be `contractAddress` to be explicit?

Reply:
I am not against it but just want to confirm since this change add lot of noise to this PR since it requires atleast 10+ files including tests.

Reply:
Not crucial in that case.